From 7646e239ed649c8a2191c5326e47caba57de5096 Mon Sep 17 00:00:00 2001
From: Martin Ritter <martin.ritter@lmu.de>
Date: Wed, 6 May 2020 09:15:33 +0200
Subject: [PATCH] Re-enable switch to C-locale when searching for C++ includes
 (ROOT-10751)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Commit df0b689 simplified the search for the C++ include directories and
tried to use a locale independent regex.

However this regex doesn't work for all locales, for example in German
the string in the output is not

```
#include <...> search starts here:
```

but the word order is switched and we have

```
Suche für »#include <...>« beginnt hier:
```

As such I propose to go back to C-locale for this query to be completely
safe against weird locale settings.
---
 interpreter/cling/lib/Interpreter/CIFactory.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/interpreter/cling/lib/Interpreter/CIFactory.cpp b/interpreter/cling/lib/Interpreter/CIFactory.cpp
index bf63c925de0..50a7497fa09 100644
--- a/interpreter/cling/lib/Interpreter/CIFactory.cpp
+++ b/interpreter/cling/lib/Interpreter/CIFactory.cpp
@@ -104,7 +104,8 @@ namespace {
                                        llvm::SmallVectorImpl<char>& Buf,
                                        AdditionalArgList& Args,
                                        bool Verbose) {
-    std::string CppInclQuery(Compiler);
+    std::string CppInclQuery("LC_ALL=C ");
+    CppInclQuery.append(Compiler);
 
     CppInclQuery.append(" -xc++ -E -v /dev/null 2>&1 |"
                         " sed -n -e '/^.include/,${' -e '/^ \\/.*++/p' -e '}'");
